src/onnx/parse_constant_of_shape.cpp

```diff
@@ -84,8 +86,16 @@ struct parse_constant_of_shape : op_parser<parse_constant_of_shape>
             l_val.visit([&](auto val) {
                 using val_type = std::remove_cv_t<typename decltype(val)::value_type>;
                 // l_val contains only one element
-                std::vector<val_type> out_vec(s.elements(), val.front());
-                l_out = literal(s, out_vec);
+                if(s.elements() > 0)
+                {
+                    std::vector<val_type> out_vec(s.elements(), val.front());
+                    l_out = literal(s, out_vec);
+                }
+                else
+                {
+                    std::vector<val_type> out_vec{val.front()};
+                    l_out = literal(s, out_vec);
+                }
```

Comment:
Why do we want to have a buffer with one value when there are no elements?

Reply:
To hold a zero value, so that every empty constant resolves to a literal with value {0}.